### PR TITLE
fix(activemodel): wire secure-password challenge to dirty._was + correct error type (P20)

### PR DESCRIPTION
**Pull request by deanmarano**
## Summary

Implements P20 from `docs/activemodel-alignment.md`, fixing audit findings #21 and #22 from `docs/activemodel/audit-lifecycle.md` §7.

**User-visible wins:**
- Challenge validation now works for records loaded from the database (previously failed silently because the WeakMap was empty for DB-loaded records)
- Too-long password emits `"passwordTooLong"` — the locale-resolvable key — instead of the generic `"too_long"` that fell through to the wrong message
- Whitespace-only digest treated as blank (password presence validation fires correctly on cache-poisoning or bare-DB migration edge cases)

## Changes

- **`secure-password.ts`**: Replace `previousDigestCache` WeakMap with `record.attributeWas(digestAttr)`. Rails `secure_password.rb:141-147` reads `attribute_digest_was` from dirty tracking; the WeakMap diverged whenever the digest was loaded via `writeFromDatabase` (no setter call). Remove `previousDigestCache` entirely — no fallback path. Simplify password setter (no pre-save logic needed). Replace `!digest` with `isBlank(digest)` for whitespace-aware presence. Emit `"passwordTooLong"` instead of `"too_long"`.
- **`i18n.ts`**: Rename `password_too_long` → `passwordTooLong` (camelCase per repo convention; matches what the validator now emits).
- **`secure-password.test.ts`**: Update two existing challenge tests to use DB-loaded record setup (construct `new User({ password_digest: existingHash })`), which is the actual scenario the tests describe. Add 5 new tests: db-loaded challenge success/failure, no-prior-digest failure, `passwordTooLong` error type, whitespace digest blank check.

**Remaining gap (out of scope):** `resetToken` option is accepted but does nothing — requires `generates_token_for` (ActiveRecord-only). Flagged in audit §7.

## Test plan

- [x] `pnpm vitest run packages/activemodel/src/secure-password.test.ts` — 56/56 ✓
- [x] `pnpm lint` — 0 errors ✓
- [x] `pnpm test:types` — 74/74 ✓
- [x] `pnpm api:compare --package activemodel` — 624/625, no regression ✓
- [x] `pnpm format:check` — clean ✓

### CODE REVIEW

File: packages/activemodel/src/i18n.ts [hunk 1]
```diff
@@ -290,7 +290,7 @@ const messages: TranslationTree = {
   empty: "can't be empty",
   not_a_date: "is not a valid date",
   required: "must exist",
-  password_too_long: "is too long",
+  passwordTooLong: "is too long",
```
[the remark this answers is not in the document]
Comment (deanmarano) [reply packages/activemodel/src/i18n.ts:293]: The rename is intentional Rails alignment. Rails emits `:password_too_long` (snake_case in Ruby) and this repo's convention is camelCase for all error type and locale keys — every other key in this file follows that pattern (e.g. `notANumber`, `notAnInteger`, `tooShort`, `tooLong`). The validator now emits `"passwordTooLong"` to match. This is a pre-release library with no published consumers; there is no backwards-compat surface to break.

File: packages/activemodel/src/secure-password.test.ts [hunk 2]
```diff
@@ -411,6 +417,62 @@ describe("SecurePasswordTest", () => {
...
+    const User = createUserClass();
+    const builder = new User({ name: "test" });
+    (builder as any).password = "secret";
+    const digest = builder.readAttribute("password_digest");
+    // No setter call on this instance — digest loaded as baseline attribute.
+    const u = new User({ name: "test", password_digest: digest });
+    (u as any).passwordChallenge = "secret";
+    expect(u.isValid()).toBe(true);
+  });
+
```
[the remark this answers is not in the document]
Comment (deanmarano) [reply packages/activemodel/src/secure-password.test.ts]: Fixed — the duplicate test (`password_challenge succeeds against db-loaded digest without setter call`) has been removed in the follow-up commit.
